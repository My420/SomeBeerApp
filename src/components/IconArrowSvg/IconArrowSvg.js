import React from 'react';
import PropTypes from 'prop-types';
import styles from './IconArrowSvg.module.scss';

const IconArrowSvg = ({ direction, containerClass, pathClass }) => {
  const containerClassName = `${styles.container} ${containerClass} ${
    styles[direction]
  }`;

  // eslint-disable-next-line no-console
  console.log('render ===== IconArrowSvg');

  return (
    <div className={containerClassName}>
      <svg viewBox="0 0 512 512" preserveAspectRatio="xMinYMin meet">
        <path
          className={pathClass}
          d="M352 128.4L319.7 96 160 256l159.7 160 32.3-32.4L224.7 256z"
        />
      </svg>
    </div>
  );
};

IconArrowSvg.propTypes = {
  direction: PropTypes.oneOf(['left', 'right', 'top', 'bottom']),
  containerClass: PropTypes.string,
  pathClass: PropTypes.string
};

IconArrowSvg.defaultProps = {
  direction: 'left',
  containerClass: styles.svgContainer,
  pathClass: styles.path
};

export default IconArrowSvg;
