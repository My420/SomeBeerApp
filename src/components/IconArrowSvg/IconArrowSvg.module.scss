@import '../../sass/variables.scss';

.container {
  overflow: hidden;
}

.container svg {
  width: 100%;
  height: 100%;
}
.left {
  transform: rotate(0deg);
}

.right {
  transform: rotate(180deg);
}

.top {
  transform: rotate(90deg);
}

.bottom {
  transform: rotate(270deg);
}

.svgContainer {
  width: 80%;
  height: 80%;
}

.path:hover {
  fill: $menu-hover-color;
}

.path:active {
  opacity: 0.5;
}
