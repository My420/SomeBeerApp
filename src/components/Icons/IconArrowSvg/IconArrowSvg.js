import React from 'react';
import PropTypes from 'prop-types';
import styles from './IconArrowSvg.module.scss';

const IconArrowSvg = ({ direction, containerClass, pathClass }) => {
  const containerClassName = `${containerClass} ${styles[direction]}`;
  return (
    <div className={containerClassName}>
      <svg width="100%" height="100%" viewBox="0 0 24 24">
        <path
          className={pathClass}
          d="M5 3l3.057-3 11.943 12-11.943 12-3.057-3 9-9z"
        />
      </svg>
    </div>
  );
};

IconArrowSvg.propTypes = {
  direction: PropTypes.oneOf(['left', 'right', 'top', 'bottom']),
  containerClass: PropTypes.string,
  pathClass: PropTypes.string
};

IconArrowSvg.defaultProps = {
  direction: 'right',
  containerClass: styles.container,
  pathClass: styles.path
};

export default IconArrowSvg;
