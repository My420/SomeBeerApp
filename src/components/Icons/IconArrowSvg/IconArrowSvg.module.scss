.container {
  display: block;
  width: 100%;
  height: 100%;
}

.path {
  fill: black;
}

.left {
  transform: rotate(180deg);
}

.right {
  transform: rotate(0deg);
}

.top {
  transform: rotate(-90deg);
}

.bottom {
  transform: rotate(90deg);
}
