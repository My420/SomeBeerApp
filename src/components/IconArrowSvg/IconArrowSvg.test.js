import React from 'react';
import { shallow } from 'enzyme';
import IconArrowSvg from './IconArrowSvg';

describe('Test <IconArrowSvg /> component', () => {
  describe('Test <IconArrowSvg /> without props', () => {
    const wrapper = shallow(<IconArrowSvg />);

    test('should correct render component without props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render div with default className', () => {
      expect(wrapper.find('div').prop('className')).toBe('container left');
    });

    test("should  render svg'", () => {
      expect(wrapper.find('svg')).toHaveLength(1);
    });

    test('svg should  have path with default className', () => {
      expect(wrapper.find('path').prop('className')).toBe('path');
    });
  });

  describe('Test <IconArrowSvg /> with props', () => {
    const wrapper = shallow(
      <IconArrowSvg
        direction="top"
        containerClass="testContainer"
        pathClass="testPath"
      />
    );

    test('should correct render component with props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  correct set className at div ', () => {
      expect(wrapper.find('div').prop('className')).toBe('testContainer top');
    });

    test('should  correct set className at path', () => {
      expect(wrapper.find('path').prop('className')).toBe('testPath');
    });
  });
});
