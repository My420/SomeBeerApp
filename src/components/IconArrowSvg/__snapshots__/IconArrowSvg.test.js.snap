// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <IconArrowSvg /> component Test <IconArrowSvg /> with props should correct render component with props 1`] = `
<div
  className="testContainer top"
>
  <svg
    height="100%"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      className="testPath"
      d="M5 3l3.057-3 11.943 12-11.943 12-3.057-3 9-9z"
    />
  </svg>
</div>
`;

exports[`Test <IconArrowSvg /> component Test <IconArrowSvg /> without props should correct render component without props 1`] = `
<div
  className="container right"
>
  <svg
    height="100%"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      className="path"
      d="M5 3l3.057-3 11.943 12-11.943 12-3.057-3 9-9z"
    />
  </svg>
</div>
`;
